### DISCUSSION TITLE
Is there a way to know how many requests are currently being served?

### DISCUSSION THREAD

**Discussion by emad-jamal** [dropwizard/dropwizard#6486] [Q&A] [answered]
Hi.. I am asking this because I need to shutdown some pubsub queues only after all requests being served are completed (Some request might consume or produce messages). Right now I implemented a filter to count those requests and here is its code:

public class CurrentRequestCountFilter implements ContainerRequestFilter, ContainerResponseFilter {

    private final AtomicInteger currentRequestCount;

    @Inject
    public CurrentRequestCountFilter(AtomicInteger currentRequestCount) {
        this.currentRequestCount = currentRequestCount;
    }

    public void filter(ContainerRequestContext containerRequestContext) {
        int currentCount = currentRequestCount.incrementAndGet();
    }

    @Override
    public void filter(
            ContainerRequestContext containerRequestContext,
            ContainerResponseContext containerResponseContext) {
        int currentCount = currentRequestCount.decrementAndGet();
    }
}

**Comment by zUniQueX** [comment, marked as answer]
Hi @emad-jamal. You can use the Dropwizard Metrics project for that.

The metrics for Jetty provide a `Counter` for the [active requests](https://github.com/dropwizard/metrics/blob/release/4.2.x/metrics-jetty11/src/main/java/io/dropwizard/metrics/jetty11/InstrumentedHandler.java#L71).
